luci: "no such package"

### Is there an existing issue for this?

- [x] I have searched among all existing issues (including closed issues)

### screenshots or captures

_No response_

### Actual behaviour

I'm getting `luci (no such package) ` when trying to run `apk add luci` on the latest openwrt release with packages for `riscv_generic`

A month ago it was working just fine, could it be related to https://github.com/openwrt/packages/pull/26889 ?



### Expected behaviour

expected behaviour would be to just install the luci package "suite"

### Steps to reproduce

- boot device
- run `apk upgrade`
- run `apk add luci`

### Additional Information

```shell
build info

root@OpenWrt:~# cat /etc/os-release 
NAME="OpenWrt"
VERSION="SNAPSHOT"
ID="openwrt"
ID_LIKE="lede openwrt"
PRETTY_NAME="OpenWrt SNAPSHOT"
VERSION_ID="snapshot"
HOME_URL="https://openwrt.org/"
BUG_URL="https://bugs.openwrt.org/"
SUPPORT_URL="https://forum.openwrt.org/"
FIRMWARE_URL="https://downloads.openwrt.org/"
BUILD_ID="r30109-d015a39dd3"
OPENWRT_BOARD="siflower/sf21"
OPENWRT_ARCH="riscv64_generic"
OPENWRT_TAINTS=""
OPENWRT_DEVICE_MANUFACTURER="OpenWrt"
OPENWRT_DEVICE_MANUFACTURER_URL="https://openwrt.org/"
OPENWRT_DEVICE_PRODUCT="Generic"
OPENWRT_DEVICE_REVISION="v0"
OPENWRT_RELEASE="OpenWrt SNAPSHOT r30109-d015a39dd3"
OPENWRT_BUILD_DATE="1750236395"
```

### What browsers do you see the problem on?

_No response_

### Relevant log output

```shell

```

> This might be related to https://lists.openwrt.org/pipermail/openwrt-devel/2020-February/027523.html
> and commit https://github.com/openwrt/openwrt/commit/642d568b0f0ac4924474871f58b2fd1c6c354476

Thanks for that analysis/guess.
That is the first really possible explanation that I have seen for the buildbot breakage for luci app.
The commit from @nbd168 is a week old, so matches the timeline when luci got broken in  buildbot.

```
make[3]: /usr/bin/env: Argument list too long
make[3]: *** [../../luci.mk:354: /builder/shared-workdir/build/sdk/bin/packages/aarch64_cortex-a53/luci/luci-25.200.62111~eb66bdc.apk] Error 127
```

The older solution from @cotequeiroz was targeting opkg, I think, so it is quite possible that the new commit from @nbd168 for apk has caused similar for apk.

Ok, I can reproduce in buildroot now.  Run a "normal" small build, just so you have some results that include `luci` in `bin/`, then create a ton of fake packages:
```
$ touch $(printf 'bin/packages/x86_64/luci/luci-abc-%d.apk ' $(seq 3000))
```
Subsequent to that, a clean will error out:
```
$ make package/luci/clean V=s
...
make[2]: /usr/bin/env: Argument list too long
make[2]: *** [../../luci.mk:353: luci-clean] Error 127
```

--- a/include/package-pack.mk
+++ b/include/package-pack.mk
@@ -18,10 +18,20 @@ define gen_package_wildcard
   $(1)$$(if $$(filter -%,$$(ABIV_$(1))),,[^a-z$(if $(CONFIG_USE_APK),,-)])*
 endef
 
+# 1: command and initial arguments
+# 2: arguments list
+# 3: tmp filename
+define maybe_use_xargs
+  $(if $(word 512,$(2)), \
+    $(file >$(3),$(2)) $(XARGS) $(1) < "$(3)"; rm "$(3)", \
+    $(1) $(2))
+endef
+
 # 1: package name
 # 2: candidate ipk files
 define remove_ipkg_files
-  $(if $(strip $(2)),$(SCRIPT_DIR)/ipkg-remove $(1) $(2))
+  $(if $(strip $(2)), \
+    $(call maybe_use_xargs,$(SCRIPT_DIR)/ipkg-remove $(1),$(2),$(TMP_DIR)/$(1).in))
 endef
 
 # 1: package name

